spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubReactiveAutoConfiguration.java

```diff
@@ -43,17 +47,20 @@
 @ConditionalOnProperty(value = "spring.cloud.gcp.pubsub.reactive.enabled", matchIfMissing = true)
 public class GcpPubSubReactiveAutoConfiguration {
 
-	@Bean
-	@ConditionalOnMissingBean(name = "pubSubReactiveScheduler")
-	Scheduler pubSubReactiveScheduler() {
-		return Schedulers.elastic();
-	}
+	private Scheduler defaultPubSubReactiveScheduler = Schedulers.newElastic("pubSubReactiveScheduler");
```

Comment:
A slight problem here is that if the user provides their own scheduler, we still have this scheduler around consuming resources. Maybe create it only when known to be needed?